File: frontend/types/analysis.ts
export type BoundingBox = { x1: number; y1: number; x2: number; y2: number };
export type DetectedObject = { id: string; label: string; confidence: number; box: BoundingBox };
export type WallRegion = { id: string; confidence: number; mask: string; boundingBox: BoundingBox; area: number };
export type Surface = { label: string; mask: string };
export type AnalysisResult = { image: { width: number; height: number }; objects: DetectedObject[]; walls: WallRegion[]; surfaces: Surface[] };

File: backend/app/schemas/analysis.py
from pydantic import BaseModel, Field


class ImageSize(BaseModel):
    width: int
    height: int


class BoundingBox(BaseModel):
    x1: int
    y1: int
    x2: int
    y2: int


class DetectedObject(BaseModel):
    id: str
    label: str
    confidence: float = Field(ge=0, le=1)
    box: BoundingBox


class WallRegion(BaseModel):
    id: str
    confidence: float = Field(ge=0, le=1)
    mask: str
    boundingBox: BoundingBox
    area: int


class Surface(BaseModel):
    label: str
    mask: str


class AnalysisResponse(BaseModel):
    image: ImageSize
    objects: list[DetectedObject]
    walls: list[WallRegion]
    surfaces: list[Surface]
